src/librustc/ty/util.rs

```diff
@@ -499,18 +499,19 @@ impl<'a, 'gcx, 'tcx, W> TypeVisitor<'tcx> for TypeIdHasher<'a, 'gcx, 'tcx, W>
     }
 
     fn visit_region(&mut self, r: &'tcx ty::Region) -> bool {
+        self.hash_discriminant_u8(r);
         match *r {
-            ty::ReErased => {
-                self.hash::<u32>(0);
-            }
+            ty::ReErased |
+            ty::ReStatic |
+            ty::ReEmpty => {}
```

Comment:
Nit: can you add a blank line in the `{}` block? This formatting makes it hard to see that the arms are a distinct group from the line below.